client/my-sites/plans-features-main/hooks/use-filtered-displayed-intervals.ts

```diff
@@ -0,0 +1,50 @@
+import {
+	getPlan,
+	getBillingMonthsForTerm,
+	isFreePlan,
+	URL_FRIENDLY_TERMS_MAPPING,
+	UrlFriendlyTermType,
+} from '@automattic/calypso-products';
+import { PlansIntent } from '@automattic/plans-grid-next';
+import { useMemo } from 'react';
+
+interface Props {
+	intent?: PlansIntent;
+	displayedIntervals: UrlFriendlyTermType[];
+	paidDomainName?: string;
+	productSlug?: string;
+}
+
+const useFilteredDisplayedIntervals = ( {
+	intent,
+	displayedIntervals,
+	paidDomainName,
+	productSlug,
+}: Props ) => {
+	return useMemo( () => {
+		let filteredIntervals = displayedIntervals;
+
+		// Hide interval terms that are less than the current plan's term in months
+		if ( productSlug && ! isFreePlan( productSlug ) ) {
+			const currentPlanIntervalInMonths = getBillingMonthsForTerm(
+				getPlan( productSlug )?.term || ''
+			);
+
+			filteredIntervals = displayedIntervals.filter( ( intervalType ) => {
+				const intervalInMonths = getBillingMonthsForTerm(
+					URL_FRIENDLY_TERMS_MAPPING[ intervalType ] || ''
+				);
+
+				return intervalInMonths >= currentPlanIntervalInMonths;
+			} );
+		}
+
+		if ( intent === 'plans-paid-media' && ! paidDomainName ) {
+			filteredIntervals = [ ...filteredIntervals, 'monthly' ];
+		}
```

Comment:
> Do you think it would be a better to remove the above and bring that logic here so that this logic is not displaced in 2 places?

Intuitively, I prefer the idea of each flow passing their own set of "displayedIntervals" as props to PlansFeaturesMain. If anything, I'd move usage of the `use-filtered-displayed-intervals` up one level into flow logic.

There is, however, already so much convention around transforming data based off of flow and intention in PlansFeaturesMain, so consolidating the logic into a `PlansFeaturesMain` hook makes sense.

> Can you also add a comment explaining the "Why" behind this decision possibly linking to a P2 or a slack discussion?

In the PR description I linked back to the request, but there really wasn't a "why" explained. The feature seemed like a simple request so I didn't question it too deeply. I'll go ahead and ask Isa